test/core.jl

```diff
@@ -7288,3 +7288,37 @@ function f37044(r)
 end
 r37044 = Ref37044(A37044{Int}.body)
 @test f37044(r37044)[1] === Int
+
+a37265() = 0
+b37265() = 0
+function c37265(d)
+    if d == 1
+        e = a37265
+    elseif d == 2
+        e = b37265
+    else
+        try
+        catch
+        end
+    end
+    e
+end
+@test_throws UndefVarError c37265(0)
+@test c37265(1) === a37265
+@test c37265(2) === b37265
+
+# This is much more reliable at triggering the issue.
+# Reproducible on 0.7.0+
```

Comment:
FWIW, the code below passes tests on all julia versions I have locally, 1.0, 1.2, 1.4.2, 1.5. Is it supposed to be `if 0`?